src/sentry_webhook/post_sentry_event_to_salesforce.ts

```diff
@@ -0,0 +1,162 @@
+// Copyright 2018 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as sentry from '@sentry/types';
+import * as https from 'https';
+
+// Defines the Salesforce form field names.
+interface SalesforceFormFields {
+  orgId: string;
+  recordType: string;
+  email: string;
+  description: string;
+  category: string;
+  cloudProvider: string;
+  sentryEventId: string;
+  os: string;
+  version: string;
+  type: string;
+}
+
+// Defines the Salesforce form values.
+interface SalesforceFormValues {
+  orgId: string;
+  recordType: string;
+}
+
+const SALESFORCE_DEV_HOST = 'google-jigsaw--uat.my.salesforce.com';
+const SALESFORCE_PROD_HOST = 'webto.salesforce.com';
+const SALESFORCE_PATH = '/servlet/servlet.WebToCase';
+const SALESFORCE_FORM_FIELDS_DEV: SalesforceFormFields = {
+  orgId: 'orgid',
+  recordType: 'recordType',
+  email: 'email',
+  description: 'description',
+  category: '00N3F000002Rqho',
+  cloudProvider: '00N3F000002Rqhs',
+  sentryEventId: '00N3F000002Rqhq',
+  os: '00N3F000002cLcN',
+  version: '00N3F000002cLcI',
+  type: 'type'
+};
+const SALESFORCE_FORM_FIELDS_PROD: SalesforceFormFields = {
+  orgId: 'orgid',
+  recordType: 'recordType',
+  email: 'email',
+  description: 'description',
+  category: '00N0b00000BqOA2',
+  cloudProvider: '00N0b00000BqOA7',
+  sentryEventId: '00N0b00000BqOA4',
+  os: '00N0b00000BqOfW',
+  version: '00N0b00000BqOfR',
+  type: 'type'
+};
+const SALESFORCE_FORM_VALUES_DEV: SalesforceFormValues = {
+  orgId: '00D3F000000DDDH',
+  recordType: '0123F000000MWTS',
+};
+const SALESFORCE_FORM_VALUES_PROD: SalesforceFormValues = {
+  orgId: '00D0b000000BrsN',
+  recordType: '0120b0000006e8i',
+};
+
+// Posts a Sentry event to Salesforce using predefined form data. Only posts if the event contains
+// an email address.
+export function postSentryEventToSalesforce(
+    event: sentry.SentryEvent, project: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    if (!event.user || !event.user.email) {
+      reject(new Error('Not posting event without email'));
```

Comment:
Seems like this way we'll get a lot of errors in our logs. How about extracting a function called something like `isSalesforceReport` and calling that first?

Reply:
Done in 4d60e67.